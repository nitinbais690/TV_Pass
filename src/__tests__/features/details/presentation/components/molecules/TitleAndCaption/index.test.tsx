import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { StyleSheet } from 'react-native';
import DetailsTitleAndCaption from 'features/details/presentation/components/molecules/TitleAndCaption';

jest.mock('core/presentation/hooks/use-app-colors');

describe('detailsTitleAndCaption', () => {
    const seriesTitle: string = 'Game of Thrones';
    const title: string = 'Season 1: Winter Is Coming';
    const caption: string = '2011 . 50 Minutes . Drama';
    const advisory: string = 'violence';

    test('render with default style', () => {
        const { container } = render(
            <DetailsTitleAndCaption seriesTitle={seriesTitle} title={title} caption={caption} advisory={advisory} />,
        );

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with custom style ', () => {
        const styles = StyleSheet.create({
            container: {
                padding: 16,
            },
        });

        const { container } = render(
            <DetailsTitleAndCaption
                seriesTitle={seriesTitle}
                title={title}
                caption={caption}
                advisory={advisory}
                style={styles.container}
            />,
        );

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render without series title', () => {
        const { container } = render(
            <DetailsTitleAndCaption seriesTitle={''} title={title} caption={caption} advisory={advisory} />,
        );

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });
});
