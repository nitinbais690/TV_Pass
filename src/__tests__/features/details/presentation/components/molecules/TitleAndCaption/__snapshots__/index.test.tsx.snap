// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsTitleAndCaption render with custom style  1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Black",
            "fontSize": 40,
            "fontWeight": "900",
          },
          Object {
            "color": "#ECECEC",
            "fontFamily": "ProximaNova-Black",
            "fontSize": 28,
            "fontWeight": "900",
            "textTransform": "uppercase",
          },
          undefined,
        ]
      }
    >
      Game of Thrones
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "opacity": 0.3,
          },
          Object {
            "marginTop": 23.314285714285713,
          },
        ]
      }
    >
      2011 . 50 Minutes . Drama
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "opacity": 0.3,
          },
          undefined,
        ]
      }
    >
      violence
    </Text>
  </View>
</View>
`;

exports[`detailsTitleAndCaption render with default style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Black",
            "fontSize": 40,
            "fontWeight": "900",
          },
          Object {
            "color": "#ECECEC",
            "fontFamily": "ProximaNova-Black",
            "fontSize": 28,
            "fontWeight": "900",
            "textTransform": "uppercase",
          },
          undefined,
        ]
      }
    >
      Game of Thrones
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "opacity": 0.3,
          },
          Object {
            "marginTop": 23.314285714285713,
          },
        ]
      }
    >
      2011 . 50 Minutes . Drama
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "opacity": 0.3,
          },
          undefined,
        ]
      }
    >
      violence
    </Text>
  </View>
</View>
`;

exports[`detailsTitleAndCaption render without series title 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Black",
            "fontSize": 40,
            "fontWeight": "900",
          },
          Object {
            "color": "#ECECEC",
            "fontFamily": "ProximaNova-Black",
            "fontSize": 28,
            "fontWeight": "900",
            "textTransform": "uppercase",
          },
          undefined,
        ]
      }
    >
      Season 1: Winter Is Coming
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "opacity": 0.3,
          },
          Object {
            "marginTop": 23.314285714285713,
          },
        ]
      }
    >
      2011 . 50 Minutes . Drama
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "opacity": 0.3,
          },
          undefined,
        ]
      }
    >
      violence
    </Text>
  </View>
</View>
`;
